### PR TITLE
Avoid overwrite

### PR DESCRIPTION
**Pull request by KirbyBroderick**

### CODE REVIEW

**Comment by zulissi** [comment]
You added a couple unnecessary files (like the ymls that are generated from the tmpl files). Probably better to remove those?

**Comment by KirbyBroderick** [comment]
Adds three functionalities:
 - The function "safe_cache" is now a wrapper for joblib.Memory.cache(func) that will raise a CacheOverrideError if calling the cached function would result in the cache being overwritten. This shouldn't ever trigger given the second change, but it's not inconceivable.
 - The function "joblib.memory._build_func_identifier", which determines where caches get written to inside of a <cachedir>/joblib folder, has been monkeypatched inside of catlas.dask_utils to incorporate the hashed function code into the filepath that gets written, meaning that conflicting code bases will have multiple folders.
 - The function "get_cached_func_location" can be used to find the save location of a particular cached function to assist with clearing out old caches.

(In the future, it may be necessary to figure out how to delete old caches intelligently.)

**Comment by zulissi** [comment]
Looks good to me! (once the file conflict are resolved and tests pass)

**Comment by zulissi** [comment]
I think this would also be better in another file than dask_utils, since adjusting how joblib caches doesn't really have anything to do with dask.

File: catlas/dask_utils.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 from dask_kubernetes.objects import make_pod_from_dict, clean_pod_template
 import yaml
 import dask
+from tokenize import open as open_py_source
```
Comment (zulissi) [inline catlas/dask_utils.py:14]: This is unnecessary

**Review by zulissi — APPROVED** [review]
Looks good, just the one extra line!

**Pull request merged**